core/src/main/java/xyz/gianlu/librespot/cache/CacheManager.java

```diff
@@ -29,21 +30,19 @@
     private static final long CLEAN_UP_THRESHOLD = TimeUnit.DAYS.toMillis(7);
     private static final Logger LOGGER = Logger.getLogger(CacheManager.class);
     private static final byte HEADER_TIMESTAMP = (byte) 0b11111111;
-    private final boolean enabled;
```

Comment:
This breaks the entire cache system if it's disabled.

Reply:
This will anyway cause an exception in `Session#cache()`.

Reply:
yes you are right , didn't see that connection there but its only used in 2 times in CDNManager and 1 time in AudioFileStreming and you are already handling null pointers at the most places. if i see correctly
![image](https://user-images.githubusercontent.com/2770344/61242385-d9319c80-a745-11e9-9a0d-9e00eb8cf800.png)

Reply:
we could make CacheManager Optional and than handle it in the 2 Files ?